File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = dfs(numbers, target, 0)
    return answer

def dfs(numbers, target, idx) :
    if len(numbers) == idx :
        # print(numbers)
        if sum(numbers) == target :
            return 1
        return 0
    answer=0
    answer+=dfs(numbers, target, idx+1)
    numbers[idx] *= -1
    answer+=dfs(numbers, target, idx+1)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
    var answer = 0;
    
    const searchBinary = (idx,num,total) => {
        const ele = total+numbers[idx]*num;
        if(idx === numbers.length-1) {
            if(ele === target) {
                answer+=1
            }
        } else {
            searchBinary(idx+1,1,ele)
            searchBinary(idx+1,-1,ele)
        }
    }
    
    searchBinary(0,1,0)
    searchBinary(0,-1,0)
    
    return answer;
}